security/tokens_and_compartments.rst

```diff
@@ -0,0 +1,212 @@
+      },
+      ...
+   ]
+
+Please notice that it is possible that more than one Patient matches the filter. This is intended behaviour of Firely Server, and it is up to you to configure a search parameter that is guaranteed to have unique values for each Patient if you need that.
+However you can also take advantage of it and allow access only to the patients from a certain General Practitioner, of whom you happen to know the Identifier::
+
+   "Filters": [
+      {
+        "FilterType": "Patient", //Filter on a Patient compartment if a 'patient' launch scope is in the auth token
+        "FilterArgument": "general-practitioner.identifier=#patient#" //... for the Patient that has an identifier matching the value of that 'patient' launch scope
+      },
+      ...
+   ]
+
+In this example the claim is still called 'patient', although it contains an Identifier of a General Practitioner. 
+This is because the CompartmentDefinition is selected by matching its code to the name of the claim, regardless of the value the claim contains.
+
+.. note::
+   Any request is scoped to the patient compartment and requests are rejected if the patient claim is not provided in the access token.
+
+.. _feature_accesscontrol_decisions:
+
+Access Control Decisions for Patient-level scopes
+-------------------------------------------------
+
+In this paragraph we will explain how access control decisions are made for the various FHIR interactions. For the examples assume a Patient Compartment with identifier=123 as filter.
+
+#. Search
+
+   a. Direct search on compartment type
+
+      :Request: ``GET [base]/Patient?name=fred``
+      :Type-Access: User must have read access to Patient, otherwise HTTP Status Code 403 is returned. 
+      :Compartment: If a Patient Compartment is active, the Filter from it will be added to the search, e.g. ``GET [base]/Patient?name=fred&identifier=123``
+
+   #. Search on type related to compartment
+
+      :Request: ``GET [base]/Observation?code=x89``
+      :Type-Access: User must have read access to Observation, otherwise HTTP Status Code 403 is returned. 
+      :Compartment: If a Patient Compartment is active, the links from Observation to Patient will be added to the search. In pseudo code: ``GET [base]/Observation?code=x89& (subject:Patient.identifier=123 OR performer:Patient.identifier=123)``
+
+   #. Search on type not related to compartment
+
+      :Request: ``GET [base]/Organization``
+      :Type-Access: User must have read access to Organization, otherwise HTTP Status Code 403 is returned. 
+      :Compartment: No compartment is applicable to Organization, so no further filters are applied.
+
+   #. Search with include outside the compartment
+
+      :Request: ``GET [base]/Patient?_include=Patient:organization``
+      :Type-Access: User must have read access to Patient, otherwise HTTP Status Code 403 is returned. If the user has read access to Organization, the _include is evaluated. Otherwise it is ignored.
```

Comment:
This is the old behavior. Currently, if we only have scope `user/Patient.read`, we get 403 as well. So, to make `_include` works, we have to use both `user/Patient.read` and `user/Organization.read`